### DISCUSSION TITLE
Bootstrap server does not start on barematal using ignition files.

### DISCUSSION THREAD

**Discussion by vdvelde** [okd-project/okd#1651] [General]
**Describe the bug**

Trying to boot a bootstrap server to deplay a bare metal environment on OKD 4.13. The bootstrap server bootsup wiht the ignition files downloads the containers and reboots in order to provision the bootstrap machine.

The most remarkable alert in the log file (data.log) is a couple of errors when trying to add resources to the namespace openshift-config.
```
Jul 04 12:42:07 bootstrap cluster-bootstrap[3166]: Failed to create "0000_00_cluster-version-operator_01_adminack_configmap.yaml" configmaps.v1./admin-acks -n openshift-config: namespaces "openshift-config" not found
```

**Version**
```
# /usr/local/bin/openshift-install version
/usr/local/bin/openshift-install 4.13.0-0.okd-2023-06-24-145750
built from commit f834ef0165bb5cbf70559eb43dad7e1203cd9aff
release image quay.io/openshift/okd@sha256:873c173b9ccb1f78bf2c5aa285227259c373191a722bcc3915375006903088a5
release architecture amd64
```

Ran both commands to create the ignition files
```
/usr/local/bin/openshift-install create manifests --dir work/
/usr/local/bin/openshift-install create ignition-configs --dir work/
```

**How reproducible**
100% 

**Log bundle**

```
# cat install-config.yaml
apiVersion: v1
baseDomain: testlab.local
compute:
- name: worker
  replicas: 0
controlPlane:
  hyperthreading: Enabled
  name: master
  replicas: 1
metadata:
  name: ocp4
networking:
  clusterNetwork:
  - cidr: 10.233.64.0/18
    hostPrefix: 24
  networkType: OVNKubernetes
  serviceNetwork:
  - 10.233.0.0/18
  machineNetwork:
  - cidr: 192.168.0.0/24
platform:
  none: {}
fips: false
pullSecret:  removed
sshKey: removed

```
```
# /usr/local/bin/openshift-install wait-for bootstrap-complete --log-level debug
DEBUG OpenShift Installer 4.13.0-0.okd-2023-06-24-145750
DEBUG Built from commit f834ef0165bb5cbf70559eb43dad7e1203cd9aff
INFO Waiting up to 20m0s (until 12:09PM) for the Kubernetes API at https://api.ocp4.testlab.local:6443...
DEBUG Loading Agent Config...
INFO API v1.26.4-2868+a7ee68b55354d8-dirty up
DEBUG Loading Install Config...
DEBUG   Loading SSH Key...
DEBUG   Loading Base Domain...
DEBUG     Loading Platform...
DEBUG   Loading Cluster Name...
DEBUG     Loading Base Domain...
DEBUG     Loading Platform...
DEBUG   Loading Networking...
DEBUG     Loading Platform...
DEBUG   Loading Pull Secret...
DEBUG   Loading Platform...
DEBUG Using Install Config loaded from state file
INFO Waiting up to 30m0s (until 12:19PM) for bootstrapping to complete...
```

[data.log](https://github.com/okd-project/okd/files/11948600/data.log)

**Comment by vrutkovs** [comment]
Please attach [log bundle](https://docs.okd.io/latest/installing/installing-troubleshooting.html#installation-bootstrap-gather_installing-troubleshooting)

**Reply by PA-MOREY** [reply]
Having the same issue while trying single node install.
[sno-test.log.gz](https://github.com/okd-project/okd/files/11949326/sno-test.log.gz)

**Reply by PA-MOREY** [reply]
Hi @vrutkovs 

Here is the log bundle:
[log-bundle-20230704151748.tar.gz](https://github.com/okd-project/okd/files/11949694/log-bundle-20230704151748.tar.gz)

**Reply by vrutkovs** [reply]
>Pulling quay.io/openshift-release-dev/ocp-release...

You're installing OCP

**Reply by PA-MOREY** [reply]
There is no documentation to install OKD on a single node. Is it possible for OKD?

**Reply by vrutkovs** [reply]
https://www.okd.io/guides/sno/

**Reply by nifty-cz** [reply]
This is describe single node installation. https://medium.com/swlh/guide-okd-4-5-single-node-cluster-832693cb752b

**Comment by vdvelde** [comment]
hi,
@vrutkovs , here is the log bundle ...

[log-bundle-20230704143726.tar.gz](https://github.com/okd-project/okd/files/11949889/log-bundle-20230704143726.tar.gz)

**Reply by vrutkovs** [reply]
No logs from master - did it boot? Can you ssh on it?

**Reply by vdvelde** [reply]
I do not have any master installed.
this error occurs when booting the bootstrap server

**Reply by vrutkovs** [reply]
This won't work if you have just one machine, you need bootstrap+master using this procedure.

If you only have one machine you need OKD-enabled assisted installer - https://github.com/openshift/assisted-service/tree/master/deploy/podman#okd-configuration and this guide - https://craig-robinson.medium.com/simple-minipc-openshift-4-10-single-node-cluster-ec6cd3c65dbc

**Comment by vdvelde** [comment]
It is not the intention to install an OKD on 1 hardware but to use 3 master nodes combined with the compute.
the logs are now also generated with the master nodes in there. Master nodes are booted, I hope you have the needed info here.
[log-bundle-20230704161554.tar.gz](https://github.com/okd-project/okd/files/11950660/log-bundle-20230704161554.tar.gz)

**Reply by vrutkovs** [reply]
I see three master IPs mentioned there, but no info from them.

They seem to have fetched Ignition from bootstrap, so that part works:
>I0704 15:09:32.784181       1 api.go:109] Pool master requested by address:"192.168.0.6:53694" User-Agent:"Ignition/2.15.0" Accept-Header: "application/vnd.coreos.ignition+json;version=3.4.0, */*;q=0.1"

Can you ssh on them? If yes, see this section: https://docs.okd.io/latest/installing/installing-troubleshooting.html#installation-manually-gathering-logs-without-SSH_installing-troubleshooting

Your bootstrap has slow disks for etcd:
>{"level":"warn","ts":"2023-07-04T14:57:48.842Z","caller":"etcdserver/util.go:166","msg":"apply request took too long","took":"703.728618ms","expected-duration":"200ms",...}

**Comment by PA-MOREY** [comment]
Hi Vadim,

Thanks a lot for the links; I could create an OKD-enabled assisted
installer and use it to download an ISO and boot my vm from it. However, it
still tries to use OCP images and can't as I'm using fake auth.
 I will try to figure out why, but if you know it handy, it would help :)

Regards,
PA

On Tue, Jul 4, 2023 at 8:56 PM Vadim Rutkovsky ***@***.***>
wrote:

> I see three master IPs mentioned there, but no info from them.
>
> They seem to have fetched Ignition from bootstrap, so that part works:
>
> I0704 15:09:32.784181 1 api.go:109] Pool master requested by address:"
> 192.168.0.6:53694" User-Agent:"Ignition/2.15.0" Accept-Header:
> "application/vnd.coreos.ignition+json;version=3.4.0, */*;q=0.1"
>
> Can you ssh on them? If yes, see this section:
> https://docs.okd.io/latest/installing/installing-troubleshooting.html#installation-manually-gathering-logs-without-SSH_installing-troubleshooting
>
> Your bootstrap has slow disks for etcd:
>
> {"level":"warn","ts":"2023-07-04T14:57:48.842Z","caller":"etcdserver/util.go:166","msg":"apply
> request took too
> long","took":"703.728618ms","expected-duration":"200ms",...}
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/okd-project/okd/discussions/1651#discussioncomment-6356081>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A74PQTW65FTOWBPA5QVZU7LXORROPANCNFSM6AAAAAAZ5VZUG4>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

Disclaimer

The information contained in this communication from the sender is confidential. It is intended solely for use by the recipient and others authorized to receive it. If you are not the recipient, you are hereby notified that any disclosure, copying, distribution or taking action in relation of the contents of this information is strictly prohibited and may be unlawful.

This email has been scanned for viruses and malware, and may have been automatically archived by Mimecast Ltd, an innovator in Software as a Service (SaaS) for business. Providing a safer and more useful place for your human generated data. Specializing in; Security, archiving and compliance. To find out more visit the Mimecast website.

**Comment by PA-MOREY** [comment]
Hi Vadim,

Thanks, I was able to deploy, however, I have an issue on my assisted
installer, on both Firefox and Chrome on a MacOS, I can't make use of "copy
to Clipboard", the button doesn't work. Up until now it was not a big deal,
but for the kubeadmin secret, it is a bit less easy, and I don't know where
to get that password from.

[image: Screenshot 2023-07-06 at 20.20.50.png]
Button is inactive, web console error:

Uncaught TypeError: navigator.clipboard is undefined
    um ClipboardCopy.js:13
    onCopy ClusterCredentials.tsx:107
    onClick ClipboardCopy.js:72
    React 11
        Kge
        Xge
        Zge
        mH
        gJ
        bJ
        GQ
        qge
        bJ
        R7
        A7
    unstable_runWithPriority scheduler.production.min.js:18
    React 3
        rm
        UQ
        dme
ClipboardCopy.js:13:4
    um ClipboardCopy.js:13
    onCopy ClusterCredentials.tsx:107
    onClick ClipboardCopy.js:72
    React 11
        Kge
        Xge
        Zge
        mH
        gJ
        bJ
        GQ
        qge
        bJ
        R7
        A7
    unstable_runWithPriority scheduler.production.min.js:18
    React 3
        rm
        UQ
        dme

PA


On Wed, Jul 5, 2023 at 12:10 PM Pierre-André Morey <
***@***.***> wrote:

> Hi Vadim,
>
> Thanks a lot for the links; I could create an OKD-enabled assisted
> installer and use it to download an ISO and boot my vm from it. However, it
> still tries to use OCP images and can't as I'm using fake auth.
>  I will try to figure out why, but if you know it handy, it would help :)
>
> Regards,
> PA
>
> On Tue, Jul 4, 2023 at 8:56 PM Vadim Rutkovsky ***@***.***>
> wrote:
>
>> I see three master IPs mentioned there, but no info from them.
>>
>> They seem to have fetched Ignition from bootstrap, so that part works:
>>
>> I0704 15:09:32.784181 1 api.go:109] Pool master requested by address:"
>> 192.168.0.6:53694" User-Agent:"Ignition/2.15.0" Accept-Header:
>> "application/vnd.coreos.ignition+json;version=3.4.0, */*;q=0.1"
>>
>> Can you ssh on them? If yes, see this section:
>> https://docs.okd.io/latest/installing/installing-troubleshooting.html#installation-manually-gathering-logs-without-SSH_installing-troubleshooting
>>
>> Your bootstrap has slow disks for etcd:
>>
>> {"level":"warn","ts":"2023-07-04T14:57:48.842Z","caller":"etcdserver/util.go:166","msg":"apply
>> request took too
>> long","took":"703.728618ms","expected-duration":"200ms",...}
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/okd-project/okd/discussions/1651#discussioncomment-6356081>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/A74PQTW65FTOWBPA5QVZU7LXORROPANCNFSM6AAAAAAZ5VZUG4>
>> .
>> You are receiving this because you commented.Message ID:
>> ***@***.***>
>>
>

Disclaimer

The information contained in this communication from the sender is confidential. It is intended solely for use by the recipient and others authorized to receive it. If you are not the recipient, you are hereby notified that any disclosure, copying, distribution or taking action in relation of the contents of this information is strictly prohibited and may be unlawful.

This email has been scanned for viruses and malware, and may have been automatically archived by Mimecast Ltd, an innovator in Software as a Service (SaaS) for business. Providing a safer and more useful place for your human generated data. Specializing in; Security, archiving and compliance. To find out more visit the Mimecast website.

**Comment by nifty-cz** [comment]
> There is no documentation to install OKD on a single node. Is it possible for OKD?

single node installation: https://medium.com/swlh/guide-okd-4-5-single-node-cluster-832693cb752b
Here much better multinode installation you can easy make single for OKD 4.13: https://computingforgeeks.com/deploy-multi-node-okd-cluster-using-fedora-coreos/?expand_article=1

I success with both, but the second one is much more better.

I see you try achieve with OVNKubernetes networktype. I never success, but OpenShiftSDN no issue so far.
